# Genomic convergence of worldwide root plants in microbe-rich soils
**Abstract**  
Convergent and/or parallel evolution is the repeated occurrence of similar adaptation in unrelated species owing to similar habitats. Darwin thought it occurred on very small scales, while increasing studies provided evidence of adaptive convergence or parallelism. However, the iconic studies generally focused on just one or a relatively small number of species shaped by specific or exclusive biogeographical regions. So far, it is unclear whether and the extent to which worldwide species can adapt to different biogeographical regions that have similar but also varying ecoenvironmental factors in convergent manner. Here we show that, despite contrasting variables across the globe, similar microbial pressure in soils confers genomic convergence in worldwide root plants. Compared to non-root plants, root plants especially those with sophisticated root systems have substantial innovative genes arising from tandem duplication (TD). Ecoenvironmental association analysis reveals soil microbial selection being the most global driver affecting TD frequency. By TD, many resistance/defense- and detoxification-related genes are expanded independently in different root plant lineages, resulting in convergent adaptation to their living microbe-rich soils. Conversely, habitat changes of root plants to reduced microbial pressure cause a decreasing TD frequency and a shrinkage of these genes. Our results demonstrate the power of soil microbial pressure in shaping genomes of root plants, providing the most worldwide prevalence of molecular convergence to date in nature.

# Data description

**1. 205species**  
The details of 205 studied genomes, including 45 non-root plants and 160 root plants. This folder includes TD-, WGD- and HGT-derived genes, gene families, and *Ks* values of TD-derived gene pairs in each species.

**2. Ancestral_and_specific_TD_events**  
Ancestral and independent TD events in *Arabidopsis thaliana*, *Schrenkiella parvula*, *Oryza sativa* and *Zizania latifolia*. The TD events that occurred before and after the divergence of *A. thaliana* and *S. parvula*, as well as *O. sativa* and *Z. latifolia*, were considered ancestral and specific TD events, respectively.

**3. Biotic_stress_responsive_genes**  
List of upregulated genes in *Arabidopsis thaliana* and *Oryza sativa* and under biotic stresses, such as *Botrytis cinereal*, *Colletotrichum tofieldiae*, flg22, *Heterodera schachti* nematodes, *Heterodera schachtii* parasitism, *Pseudomonas syringae*, *Rhizobium* sp, and *Verticillium dahlia* on *Arabidopsis thaliana* and *Meloidogyne incognita*, *Magnaporthe oryzae*, *Azospirillum brasilense*, Soil microbiome, *Hirschmanniella oryzae*, *Rhizophagus irregularis*, *Burkholderia vietnamiensis* and *Paraburkholderia kururiensis* on *Oryza sativa*. 

**4. GLM_analysis**  
The details of the species geographical coordinates, climatic, anthropogenic, edaphic, and plant morpho-ecospace variables, and the GLM analysis results of 1000 bootstrap resampling replicates.

**5. GO_enrichment**  
The GO term annotations for 28 representative species and enriched terms of TD- and WGD-derived family genes.

**6. MLtree**  
The original phylogenetic tree files for the ten TD-derived families. The phylogenetic trees were constructed using 17 representative plants, including Rhodophytes (*Galdieria sulphuraria* and *Gracilariopsis chorda*), Prasinodermophyta (*Prasinoderma coloniale*), Chlorophytes (*Chlamydomonas reinhardtii* and *Chloropicon primus*), Charophytes (*Klebsormidium nitens* and *Mesotaenium endlicherianum*), Bryophytes (*Marchantia polymorpha* and *Anthoceros angustus*), Lycophytes (*Isoetes taiwanensis* and *Selaginella moellendorffii*), Ferns (*Adiantum capillus-veneris* and *Alsophila spinulosa*), Gymnosperms (*Torreya grandis* and *Pinus lambertiana*), and Angiosperms (*Arabidopsis thaliana* and *Oryza sativa*). 

**7. Mangroves**  
The details of 15 mangrove genomes. You can obtain gene lists of TD-derived genes, gene families, and *Ks* values of TD-gene pairs in each mangrove species.

**8. Code**  
The code used in this study, including 'HGTdetector.sh' and other scripts. 
